loop_stack.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+set -eux
+
+# root directory for the repository
+export HPC_STACK_ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+usage() {
+  set +x
+  echo
+  echo "Usage: $0 -p <prefix> | -c <config> | -y <yaml> -h"
+  echo
+  echo "  -p  installation prefix <prefix>    DEFAULT: $HOME/opt"
+  echo "  -c  use configuration file <config> DEFAULT: config/config_custom.sh"
+  echo "  -y  use yaml file <yaml>            DEFAULT: config/stack_custom.yaml"
+  echo "  -h  display this message and quit"
+  echo
+  exit 1
+}
+
+# ==============================================================================
+
+[[ $# -eq 0 ]] && usage
+
+# Defaults:
+export PREFIX="$HOME/opt"
+config="${HPC_STACK_ROOT}/config/config_custom.sh"
+
+while getopts ":p:c:y:h" opt; do
+  case $opt in
+    p)
+      export PREFIX=$OPTARG
+      ;;
+    c)
+      config=$OPTARG
+      ;;
+    y)
+      single_yaml=$OPTARG
+      ;;
```

Comment:
Building MPI with GCC and not with Intel

Reply:
It does seem a little misplaced.

In my previous iteration of this idea I head an option to build MPI based on the version specified in the config rather than using the stack file. It seems repetitive to specify building MPI in the stack file and also specifying it in the config file. All the information regarding MPI flavor/version is available in the config. You just need to know whether to build it or not.

It also leads to the possibility of mismatching them by specifying it in two. You could change what MPI version you're building in the stack file, but not in the config (which I've done before).

Reply:
I removed it. After thinking about it, I don't think it's necessary.